component/src/main/java/org/wso2/carbon/identity/authenticator/FoursquareAuthenticator.java

```diff
@@ -30,322 +30,367 @@
+		HttpURLConnection httpConnection = null;
+		JSONObject jsonObj = null;
+		try {
+			URL obj = new URL(url + "&" + FoursquareAuthenticatorConstants.FOURSQUARE_OAUTH2_ACCESS_TOKEN_PARAMETER +
+			                  "=" + accessToken);
+			URLConnection connection = obj.openConnection();
+			// Cast to a HttpURLConnection
+			if (connection instanceof HttpURLConnection) {
+				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setConnectTimeout(FoursquareAuthenticatorConstants.CONNECTION_TIMEOUT_VALUE);
+				httpConnection.setReadTimeout(FoursquareAuthenticatorConstants.READ_TIMEOUT_VALUE);
+				httpConnection.setRequestMethod(FoursquareAuthenticatorConstants.HTTP_GET_METHOD);
+				bufferedReader = new BufferedReader(new InputStreamReader(httpConnection.getInputStream()));
+			} else {
+				throw new ApplicationAuthenticatorException("Exception while casting the HttpURLConnection");
+			}
+			String inputLine = bufferedReader.readLine();
+			while (inputLine != null) {
+				jsonResponseCollector.append(inputLine).append("\n");
+				inputLine = bufferedReader.readLine();
+			}
+			jsonObj = new JSONObject(jsonResponseCollector.toString());
+		} catch (MalformedURLException e) {
+			throw new ApplicationAuthenticatorException(
+					"MalformedURLException while generating the user info URL: " + url, e);
+		} catch (ProtocolException e) {
+			throw new ApplicationAuthenticatorException("ProtocolException while setting the request method: " +
+			                                            FoursquareAuthenticatorConstants.HTTP_GET_METHOD +
+			                                            " for the URL: " + url, e);
+		} catch (IOException e) {
+			throw new ApplicationAuthenticatorException("Error when reading the response from " + url +
+			                                            "to update user claims", e);
+		} finally {
+			IdentityIOStreamUtils.closeReader(bufferedReader);
+			if (httpConnection != null) {
+				httpConnection.disconnect();
+			}
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Receiving the response for the User info: " + jsonResponseCollector.toString());
+		}
+		return jsonObj;
+	}
+
+	/**
+	 * Get the Foursquare specific claim dialect URI.
+	 *
+	 * @return Claim dialect URI.
+	 */
+	@Override
+	public String getClaimDialectURI() {
+		return FoursquareAuthenticatorConstants.CLAIM_DIALECT_URI;
+	}
+
+	/**
+	 * This method is to build the claims for the user info.
+	 *
+	 * @param token                   token
+	 * @param authenticatorProperties authenticatorProperties
+	 * @return claims
+	 */
+	private Map<ClaimMapping, String> buildClaims(OAuthClientResponse token,
+	                                              Map<String, String> authenticatorProperties)
+			throws ApplicationAuthenticatorException {
+		Map<ClaimMapping, String> claims = new HashMap<>();
+		String accessToken = token.getParam("access_token");
+		String url = getUserInfoEndpoint(token, authenticatorProperties);
+		JSONObject userData;
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-foursquare/pull/2/commits/58520246ef2e797650ce96df5d6f5fbde73cfa2d